File: modules/wem-api/src/main/java/com/enonic/wem/api/query/aggregation/TermsAggregation.java
package com.enonic.wem.api.query.aggregation;

public class TermsAggregation
    extends Aggregation
{
    private final Buckets buckets;

    public TermsAggregation( final Builder builder )
    {
        super( builder );
        this.buckets = builder.buckets;
    }

    public Buckets getBuckets()
    {
        return buckets;
    }

    public static class Builder
        extends Aggregation.Builder<Builder>
    {
        private Buckets buckets;

        public Builder buckets( final Buckets buckets )
        {
            this.buckets = buckets;
            return this;
        }

        public TermsAggregation build()
        {
            return new TermsAggregation( this );
        }
    }

}

File: modules/wem-webapp/src/main/webapp/admin/common/js/query/aggregation/TermsAggregation.ts
module api.query.aggregation {

    export class TermsAggregation extends Aggregation {

        private buckets:Bucket[];

        constructor( builder:TermsAggregationBuilder )
        {
            super( builder );
            this.buckets = builder.buckets;
        }

        public getBuckets():Bucket[]
        {
            return this.buckets;
        }
    }

    export class TermsAggregationBuilder extends AggregationBuilder {

        buckets:Bucket[] = [];

        public setBuckets( buckets:Bucket[] ):TermsAggregationBuilder
        {
            this.buckets = buckets;
            return this;
        }

        public addBucket( bucket:Bucket ):TermsAggregationBuilder
        {
            this.buckets.push( bucket );
            return this;
        }

        public build():TermsAggregation
        {
            return new TermsAggregation( this );
        }
    }
}
